tests/frame_submission.cpp

```diff
@@ -144,3 +145,26 @@ TEST_F(FrameSubmission, when_client_endlessly_requests_frame_then_callbacks_are_
 
     EXPECT_THAT(std::chrono::steady_clock::now(), Lt(timeout)) << "Timed out looping in frame callback storm";
 }
+
+/*
+ * The above test checks that frame callbacks are throttled, but not that
+ * they are sent at all.
+ *
+ * We also need to test that these type of buffer-less frame requests
+ * *do* get a callback, eventually.
+ */
+TEST_F(FrameSubmission, frame_requests_without_buffer_are_called_back_eventually)
+{
+    std::array<bool, 5> called = {0};
```

Comment:
Relying on the implicit conversion to `bool` seems ugly
```suggestion
    std::array<bool, 5> called{false};
```

Reply:
I think our code is being misleading in both cases; this is aggregate initialisation, right? Which is going to initialised the *first* member with the value we provide, then default-initialise the rest of the array (conveniently, in this case, to `false`)?